cmd/common/containerized-gadgets/build.go

```diff
@@ -0,0 +1,231 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package containerizedgadgets
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/opencontainers/image-spec/specs-go"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/spf13/cobra"
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/content"
+)
+
+type buildOptions struct {
+	progAmd64FilePath  string
+	progArm64FilePath  string
+	definitionFilePath string
+	tag                string
+}
+
+func NewBuildCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "build tag",
+		Short: "build a containerized gadget image",
+		RunE:  runBuild,
+	}
+
+	cmd.Flags().String("prog-amd64", "", "TODO: FILEPATH: eBPF program")
+	cmd.Flags().String("prog-arm64", "", "TODO: FILEPATH: eBPF program")
+	cmd.Flags().String("definition", "", "TODO: FILEPATH: definition file")
+	cmd.Flags().String("tag", "", "TODO: tag to push")
+
+	return cmd
+}
+
+func runBuild(cmd *cobra.Command, args []string) error {
+	if len(args) != 1 {
+		return fmt.Errorf("expected exactly one unnamed argument")
+	}
+	o := &buildOptions{
+		progAmd64FilePath:  cmd.Flag("prog-amd64").Value.String(),
+		progArm64FilePath:  cmd.Flag("prog-arm64").Value.String(),
+		definitionFilePath: cmd.Flag("definition").Value.String(),
+		tag:                args[0],
+	}
+
+	if o.progArm64FilePath == "" && o.progAmd64FilePath == "" {
+		return fmt.Errorf("at least one of --prog-amd64 or --prog-arm64 must be specified")
+	}
+
+	ociStore, err := getLocalOciStore()
+	if err != nil {
+		return fmt.Errorf("get oci store: %w", err)
+	}
+
+	indexDesc, err := createImageIndex(ociStore, o)
+	if err != nil {
+		return fmt.Errorf("create image index: %w", err)
+	}
+
+	err = ociStore.Tag(context.Background(), *indexDesc, o.tag)
+	if err != nil {
+		return fmt.Errorf("tag manifest: %w", err)
+	}
+
+	fmt.Printf("Successfully built %s@%s\n", o.tag, indexDesc.Digest)
+	return nil
+}
+
+func pushDescriptorIfNotExists(target oras.Target, desc ocispec.Descriptor, contentReader io.Reader) error {
+	exists, err := target.Exists(context.Background(), desc)
+	if err != nil {
+		return fmt.Errorf("check descriptor exists: %w", err)
+	}
+	if exists {
+		return nil
+	}
+	err = target.Push(context.Background(), desc, contentReader)
+	if err != nil {
+		return fmt.Errorf("push descriptor: %w", err)
+	}
+
+	return nil
+}
+
+func createEbpfDescAndPush(target oras.Target, progFilePath string) (*ocispec.Descriptor, error) {
```

Comment:
Nit: Many functions are returning a pointer but it's really not useful because it seems all oras functions receive the object, so I think it's better to return the structure to avoid the use of `*` and `&` all around.

Reply:
It was either returning `nil` + dealing with dereferencing or returning `ocispec.Descriptor{}`. I changed it, since I don't know which way is the better one :D